packs/cis_azure_pack.sql

```diff
@@ -74,3 +74,192 @@ WHERE 1=1
+         'subscription_id', subscription_id
+       ) AS identity
+FROM (
+  SELECT DISTINCT
+    tenant_id,
+    subscription_id,
+    name,
+    type,
+    properties:pricingTier::STRING pricing_tier
+  FROM data.azure_collect_pricings
+  WHERE recorded_at > CURRENT_DATE - 2
+)
+WHERE 1=1
+  AND pricing_tier != 'Standard'
+;
+
+CREATE OR REPLACE VIEW rules.AZURE_CIS_2_2_VIOLATION_QUERY COPY GRANTS
+  COMMENT='Automatic provisioning of monitoring agent should be "On"
+  @id I9QOIRZ53QG
+  @tags cis, azure, security-center'
+AS
+SELECT 'I9QOIRZ53QG' AS query_id
+     , 'Azure CIS 2.2: Ensure that "Automatic provisioning of monitoring agent" is set to "On"' AS title
+     , OBJECT_CONSTRUCT(
+         'cloud', 'azure',
+         'account', tenant_id
+       ) AS environment
+     , (
+         'Subscription `' || subscription_id || '`, ' ||
+         'in tenant `' || tenant_id || '`, ' ||
+         'named "' || name || '"'
+       ) AS object
+     , 'Subscription in violation of AZ CIS 2.2: ' || object AS description
+     , CURRENT_TIMESTAMP AS alert_time
+     , OBJECT_CONSTRUCT(*) AS event_data
+     , 'SnowAlert' AS detector
+     , 'High' AS severity
+     , 'devsecops' AS owner
+     , OBJECT_CONSTRUCT(
+         'query_id', query_id,
+         'cloud', 'azure',
+         'tenant_id', tenant_id,
+         'subscription_id', subscription_id
+       ) AS identity
+FROM (
+  SELECT DISTINCT
+    tenant_id,
+    subscription_id,
+    name,
+    type,
+    properties:autoProvision::STRING auto_provision
+  FROM data.azure_collect_auto_provisioning_settings
+  WHERE recorded_at > CURRENT_DATE - 2
+)
+WHERE 1=1
+  AND auto_provision != 'On'
+;
+
+CREATE OR REPLACE VIEW rules.AZURE_CIS_2_3_TO_15_VIOLATION_QUERY COPY GRANTS
+  COMMENT='Various ASC Default policy settings should not be "Disabled"
+  @id 6QL7YIUFM6L
+  @tags cis, azure, security-center'
+AS
+SELECT '6QL7YIUFM6L' AS query_id
+     , 'Azure CIS 2.3 to 2.15: Ensure various ASC Default policy settings are not "Disabled"' AS title
+     , OBJECT_CONSTRUCT(
+         'cloud', 'azure',
+         'account', tenant_id
+       ) AS environment
+     , (
+         'Subscription `' || subscription_id || '`, ' ||
+         'in tenant `' || tenant_id || '`, ' ||
+         'named "' || name || '"'
+       ) AS object
+     , 'Subscription in violation of AZ CIS 2.3—15: ' || object AS description
+     , CURRENT_TIMESTAMP AS alert_time
+     , OBJECT_CONSTRUCT(*) AS event_data
+     , 'SnowAlert' AS detector
+     , 'High' AS severity
+     , 'devsecops' AS owner
+     , OBJECT_CONSTRUCT(
+         'query_id', query_id,
+         'cloud', 'azure',
+         'tenant_id', tenant_id,
+         'subscription_id', subscription_id
+       ) AS identity
+FROM (
+  SELECT DISTINCT
+    tenant_id,
+    subscription_id,
+    name,
+    type,
+    sku,
+    properties:parameters params
+  FROM data.azure_collect_policy_assignments
+  WHERE recorded_at > CURRENT_DATE - 2
+)
+WHERE 1=1
+  AND 'Disabled' IN (
```

Comment:
I had a look at the raw data and I think that most of the times (except tenant:d479c7c9-2632-445a-b22d-7c19e68774f6 subscription:f4b00c5f-f6bf-41d6-806b-e1cac4f1f36f) the parameters fields is an empty dictionary, I'm not sure what it means, my guess is that this feature is disabled in the subscription. I think we would have to check with the Azure team to confirm. If that's the case I think we should trigger this on empty dictionary AND incorrect value in the dictionary field.

Probably this needs to be split into one violation per CIS entry to show up correctly in the dashboards :(

Reply:
Good catch -- the CIS doc says "Ensure the output of the below command is not Disabled or Empty", so going to go with that rather than reaching out to their support, unless you'd prefer to double check with them?

Reply:
nope, go for it!